contracts/utils/MultiRateLimited.sol

```diff
@@ -0,0 +1,272 @@
+        uint256 _rateLimitPerSecond,
+        uint256 _maxRateLimitPerSecondMRL,
+        uint256 _maxBufferCap,
+        uint256 _bufferCap,
+        bool _doPartialAction
+    )
+        CoreRef(coreAddress)
+        RateLimited(_maxRateLimitPerSecond, _rateLimitPerSecond, _bufferCap, _doPartialAction)
+    {
+        require(_maxBufferCap < _bufferCap, "MultiRateLimited: max buffer cap invalid");
+
+        maxRateLimitPerSecond = _maxRateLimitPerSecondMRL;
+        maxBufferCap = _maxBufferCap;
+    }
+
+    // ----------- Governor and Admin only state changing api -----------
+
+    /// @notice update the sub gov rate limit per second
+    /// @param newRateLimitPerSecond new maximum rate limit per second for sub governors
+    /// TODO determine the proper modifier for this function
+    function updateMaxRateLimitPerSecond(uint256 newRateLimitPerSecond) external virtual override onlyGovernor {
+        uint256 oldMaxRateLimitPerSecond = maxRateLimitPerSecond;
+        maxRateLimitPerSecond = newRateLimitPerSecond;
+
+        emit MultiMaxRateLimitPerSecondUpdate(oldMaxRateLimitPerSecond, newRateLimitPerSecond);
+    }
+
+    /// @notice update the sub gov max buffer cap
+    /// @param newSubGovBufferCap new buffer cap for sub governor added addresses
+    /// TODO determine the proper modifier for this function
+    function updateMaxBufferCap(uint256 newSubGovBufferCap) external virtual override onlyGovernor {
+        uint256 oldBufferCap = maxBufferCap;
+        maxBufferCap = newSubGovBufferCap;
+
+        emit MultiBufferCapUpdate(oldBufferCap, newSubGovBufferCap);
+    }
+
+    /// @notice add an authorized rateLimitedAddress contract
+    /// @param rateLimitedAddress the new address to add as a rateLimitedAddress
+    /// @param _rateLimitPerSecond the rate limit per second for this rateLimitedAddress
+    /// @param _bufferCap  the buffer cap for this rateLimitedAddress
+    function addAddress(
+        address rateLimitedAddress,
+        uint112 _rateLimitPerSecond,
+        uint144 _bufferCap
+    ) external virtual override onlyGovernor {
+        _addAddress(rateLimitedAddress, _rateLimitPerSecond, _bufferCap);
+    }
+
+    /// @notice add an authorized rateLimitedAddress contract
+    /// @param rateLimitedAddress the address whose buffer and rate limit per second will be set
+    /// @param _rateLimitPerSecond the new rate limit per second for this rateLimitedAddress
+    /// @param _bufferCap  the new buffer cap for this rateLimitedAddress
+    function updateAddress(
+        address rateLimitedAddress,
+        uint112 _rateLimitPerSecond,
+        uint144 _bufferCap
+    ) external virtual override hasAnyOfTwoRoles(TribeRoles.MINOR_MINTER_ADD_ROLE, TribeRoles.GOVERNOR) {
+        if (core().hasRole(TribeRoles.MINOR_MINTER_ADD_ROLE, msg.sender)) {
+            require(_rateLimitPerSecond <= maxRateLimitPerSecond, "MultiRateLimited: rate limit per second exceeds non governor allowable amount");
```

Comment:
The concept of a "non governor allowable amount" is confusing. Might be better to just say "rate limit too high"

Reply:
It makes perfect sense in this context. We are saying allow you to add up to a certain amount of rate limit per second as a non governor.